### PR TITLE
[SPARK-30992][DSTREAMS] Arrange scattered config of streaming module

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
I found a lot scattered config in `Streaming`.I think should arrange these config in unified position.

### Why are the changes needed?
Arrange scattered config

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Exists UT

### CODE REVIEW

**Comment by beliefer** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
cc @HeartSaVioR and @zsxwing

File: streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala [hunk 1]
```diff
@@ -61,8 +62,7 @@ private[streaming] object StateMap {
   def empty[K, S]: StateMap[K, S] = new EmptyStateMap[K, S]
 
   def create[K: ClassTag, S: ClassTag](conf: SparkConf): StateMap[K, S] = {
-    val deltaChainThreshold = conf.getInt("spark.streaming.sessionByKey.deltaChainThreshold",
-      DELTA_CHAIN_LENGTH_THRESHOLD)
+    val deltaChainThreshold = conf.get(SESSION_BY_KEY_DELTA_CHAIN_THRESHOLD)
```
Comment (HeartSaVioR) [inline streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala]: Is `DELTA_CHAIN_LENGTH_THRESHOLD` used other than this?

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingConf.scala [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    ConfigBuilder("spark.streaming.receiver.maxRate")
+      .longConf
+      .createWithDefault(Long.MaxValue)
+
+  private[streaming] val BACKPRESSURE_INITIAL_RATE =
+    ConfigBuilder("spark.streaming.backpressure.initialRate")
+      .longConf
+      .createOptional
+
+  val BLOCK_INTERVAL =
```
Comment (HeartSaVioR) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingConf.scala]: Is there specific reason this is exposed as public?

**Review by HeartSaVioR — COMMENTED** [review]
Looks good. Only minor questions.

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingConf.scala (hunk 2 above)
Comment (beliefer) [reply streaming/src/main/scala/org/apache/spark/streaming/StreamingConf.scala]: Sorry, I forgot it.

File: streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala (hunk 1 above)
Comment (beliefer) [reply streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala]: Only here `DELTA_CHAIN_LENGTH_THRESHOLD`

Comment (HeartSaVioR) [reply streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala]: OK, then could we remove `DELTA_CHAIN_LENGTH_THRESHOLD`?

Comment (beliefer) [reply streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala]: OK.

Comment (beliefer) [reply streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala]: I'm sorry, I read it wrong.`DELTA_CHAIN_LENGTH_THRESHOLD` used in two place.
https://github.com/apache/spark/blob/5c29613428bffad5b0aae7cbcec85123a1ef4819/streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala#L87
https://github.com/apache/spark/blob/5c29613428bffad5b0aae7cbcec85123a1ef4819/streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala#L102

**Comment by gaborgsomogyi** [comment]
As I see you've created the Jira with Structured Streaming component and then trying to make DStreams modifications. I suggest to change the title to `[SPARK-30992][DSTREAMS] Arrange scattered config of streaming module` in order to avoid misunderstandings like this. Additionally changed the jira component.

**Comment by beliefer** [comment]
@gaborgsomogyi Thanks.

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingConf.scala [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+  private[streaming] val RECEIVER_MAX_RATE =
+    ConfigBuilder("spark.streaming.receiver.maxRate")
+      .longConf
+      .createWithDefault(Long.MaxValue)
+
+  private[streaming] val BACKPRESSURE_INITIAL_RATE =
+    ConfigBuilder("spark.streaming.backpressure.initialRate")
+      .longConf
+      .createOptional
```
Comment (gaborgsomogyi) [inline streaming/src/main/scala/org/apache/spark/streaming/StreamingConf.scala]: maybe `fallbackConf(RECEIVER_MAX_RATE)`?

File: streaming/src/main/scala/org/apache/spark/streaming/receiver/BlockGenerator.scala [hunk 4]
```diff
@@ -100,7 +101,7 @@ private[streaming] class BlockGenerator(
   }
   import GeneratorState._
 
-  private val blockIntervalMs = conf.getTimeAsMs("spark.streaming.blockInterval", "200ms")
+  private val blockIntervalMs = conf.get(BLOCK_INTERVAL)
   require(blockIntervalMs > 0, s"'spark.streaming.blockInterval' should be a positive value")
```
Comment (gaborgsomogyi) [inline streaming/src/main/scala/org/apache/spark/streaming/receiver/BlockGenerator.scala]: `BLOCK_INTERVAL.key`?

File: streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala (hunk 1 above)
Comment (gaborgsomogyi) [reply streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala]: Do the newly defined config's default value and the mentioned other places have to keep the same value? I mean if one change the one should the others be changed? (now the value defined in 2 distinct places)

**Comment by gaborgsomogyi** [comment]
I've just checked the documentation and `spark.streaming.receiver.maxRate` says in the doc that default number is `not set` but in the code it's ` Long.MaxValue.

File: streaming/src/main/scala/org/apache/spark/streaming/StreamingConf.scala (hunk 3 above)
Comment (beliefer) [reply streaming/src/main/scala/org/apache/spark/streaming/StreamingConf.scala]: OK. Thanks!

File: streaming/src/main/scala/org/apache/spark/streaming/receiver/BlockGenerator.scala (hunk 4 above)
Comment (beliefer) [reply streaming/src/main/scala/org/apache/spark/streaming/receiver/BlockGenerator.scala]: OK

File: streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala (hunk 1 above)
Comment (beliefer) [reply streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala]: I'm not sure about this, I hope someone will be more clear about this.

**Comment by beliefer** [comment]
> I've just checked the documentation and `spark.streaming.receiver.maxRate` says in the doc that default number is `not set` but in the code it's `Long.MaxValue`.

I don't know the reason.

File: streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala (hunk 1 above)
Comment (gaborgsomogyi) [reply streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala]: I've had a deeper look and `DELTA_CHAIN_LENGTH_THRESHOLD` is `OpenHashMapBasedStateMap` specific default value. I don't think we have to move it but I think the newly extracted configuration must use the constant.

**Comment by gaborgsomogyi** [comment]
> > I've just checked the documentation and `spark.streaming.receiver.maxRate` says in the doc that default number is `not set` but in the code it's `Long.MaxValue`.
> 
> I don't know the reason.

Had a deeper look and `Long.MaxValue` is defined as unlimited (not set) so we can consider it's fine.

File: streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala (hunk 1 above)
Comment (beliefer) [reply streaming/src/main/scala/org/apache/spark/streaming/util/StateMap.scala]: OK.

**Comment by beliefer** [comment]
@gaborgsomogyi Thanks.

**Comment by beliefer** [comment]
retest this please

**Comment by beliefer** [comment]
retest this please

**Comment by gaborgsomogyi** [comment]
retest this please

**Review by gaborgsomogyi — APPROVED** [review]
LGTM (pending tests).

**Review by HeartSaVioR — APPROVED** [review]
LGTM

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0. Thanks, @beliefer, @HeartSaVioR, @gaborgsomogyi.

**Comment by beliefer** [comment]
@HyukjinKwon @HeartSaVioR @gaborgsomogyi Thanks for all your review.

**Pull request closed without merging**